lib/ClangImporter/ImportDecl.cpp

```diff
@@ -7670,6 +7670,34 @@ static bool suppressOverriddenMethods(ClangImporter::Implementation &importer,
   return false;
 }
 
+void addCompletionHandlerAttribute(Decl *asyncImport,
+                                   ArrayRef<Decl *> members,
+                                   ASTContext &SwiftContext) {
+  auto *ayncFunc = dyn_cast_or_null<AbstractFunctionDecl>(asyncImport);
+  if (!ayncFunc)
+    return;
+
+  auto errorConvention = ayncFunc->getForeignErrorConvention();
+  auto asyncConvention = ayncFunc->getForeignAsyncConvention();
+  if (!asyncConvention)
+    return;
+
+  unsigned completionIndex = asyncConvention->completionHandlerParamIndex();
+  if (errorConvention &&
```

Comment:
I need to look into the ObjectiveC import a little more, I'm not sure that this is actually correct as is (ie. that the error parameter is always removed). I'll do that on Monday.

Reply:
It's very, very rare, but yes---if there is an error convention, the error parameter is removed.

Reply:
Thanks Doug :)